x/tokenfactory/keeper/before_send.go

```diff
@@ -121,8 +132,12 @@ func (k Keeper) callBeforeSendListener(ctx sdk.Context, from, to sdk.AccAddress,
 				return err
 			}
 
-			em := sdk.NewEventManager()
+			// if its track before send, apply gas meter to prevent infinite loop
+			if !blockBeforeSend {
+				ctx = ctx.WithGasMeter(sdk.NewGasMeter(types.TrackBeforeSendGasLimit))
```

Comment:
We want to write this consumed gas meter result to the parent gas meter though. (So if that panics, then its fine)

I think that looks like keeping parentCtx, and doing "parentCtx.GasMeter.Use(childCtx.GasMeter.GasUsed)"

Reply:
If it's because of panic catching, it's already fine as of right now since we recover thread from panic though using defer function above right? (Probably me missing something)

Reply:
Suppose the parent gas meter had say 5 million gas.

It does this + some other stuff.

We still want the gas consumed here to get added to that gas meter. Its totally fine if it can't out of gas error inside this function, the next thing that consumes gas will cause the gas panic

Reply:
Attempt in https://github.com/osmosis-labs/osmosis/pull/5927/commits/f8619ff632c03746bb1b8b92938356fd80a65770.